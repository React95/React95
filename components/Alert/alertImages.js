import dialogError from 'assets/pattern/dialog-error.png';
import dialogInfo from 'assets/pattern/dialog-info.png';
import dialogWarning from 'assets/pattern/dialog-warning.png';

export { dialogError, dialogInfo, dialogWarning };
