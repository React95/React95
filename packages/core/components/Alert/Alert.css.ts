import { style } from '@vanilla-extract/css';
import { contract } from '../themes/contract.css';

export const message = style({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
});

export const icon = style({
  paddingTop: contract.space[7],
  paddingRight: contract.space[15],
  paddingBottom: contract.space[7],
  paddingLeft: contract.space[7],
});

export const dialog = style({
  display: 'flex',
  flexDirection: 'row',
});
