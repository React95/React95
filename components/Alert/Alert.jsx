import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Modal from '../Modal';

import { dialogError, dialogInfo, dialogWarning } from './alertImages';

const Dialog = styled.div`
  display: flex;
  flex-direction: row;
`;

function imageDialog({ type }) {
  const images = {
    error: dialogError,
    info: dialogInfo,
    warning: dialogWarning,
  };

  return `background-image: url(${images[type] || images.error});`;
}

Dialog.Image = styled.div`
  ${imageDialog}
  background-repeat: no-repeat;
  background-size: 70%;
  height: 45px;
  width: 45px;
  background-position: center;
  margin-right: 10px;
`;

Dialog.Message = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

const Alert = ({
  type, title, message, onCloseModal, closeAlert,
}) => {
  const defaultPosition = {
    x: Math.floor(window.innerWidth / 2) - 150,
    y: Math.floor(window.innerHeight / 2) - 80,
  };

  return (
    <Modal
      title={title}
      closeModal={onCloseModal}
      buttonsAlignment="center"
      buttons={[{ value: 'Ok', onClick: closeAlert }]}
      defaultPosition={defaultPosition}
      width="auto"
      height={120}
    >
      <Dialog>
        <Dialog.Image type={type} />
        <Dialog.Message>{message}</Dialog.Message>
      </Dialog>
    </Modal>
  );
};

Alert.displayName = 'Alert';

Alert.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['info', 'error', 'warning']),
  closeAlert: PropTypes.func,
};

Alert.defaultProps = {
  type: 'error',
  closeAlert: () => {},
};

export default Alert;
