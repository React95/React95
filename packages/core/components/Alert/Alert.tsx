import React, { forwardRef, useEffect } from 'react';
import type { FC } from 'react';
import { User2, User3, User4, User5 } from '@react95/icons';
import * as styles from './Alert.css';

import { Modal, ModalProps } from '../Modal/Modal';

import sound from './assets/chord.mp3';

export type AlertType = 'error' | 'info' | 'question' | 'warning';

const RenderImage: FC<{ option: string }> = ({ option }) => {
  switch (option) {
    case 'info':
      return <User5 width={32} height={32} variant="32x32_4" />;
    case 'question':
      return <User3 width={32} height={32} variant="32x32_4" />;
    case 'warning':
      return <User2 width={32} height={32} variant="32x32_4" />;
    case 'error':
    default:
      return <User4 width={32} height={32} variant="32x32_4" />;
  }
};

export type AlertProps = ModalProps & {
  message: string;
  hasSound?: boolean;
  type?: AlertType;
};

export const Alert = forwardRef<HTMLDivElement, AlertProps>(
  ({ type = 'error', message, hasSound, ...rest }, ref) => {
    if (hasSound) {
      useEffect(() => {
        const audio = new Audio(sound);
        audio.play();
      }, []);
    }

    return (
      <Modal height="120" hasWindowButton={false} {...rest} ref={ref}>
        <div className={styles.dialog}>
          <div className={styles.icon}>
            <RenderImage option={type} />
          </div>
          <div className={styles.message}>{message}</div>
        </div>
      </Modal>
    );
  },
);

Alert.defaultProps = {
  type: 'error',
  buttons: [{ value: 'OK', onClick: () => {} }],
  buttonsAlignment: 'center',
  positionOffset: {
    x:
      typeof window == 'undefined'
        ? 0
        : Math.floor(window.innerWidth / 2) - 150,
    y:
      typeof window == 'undefined'
        ? 0
        : Math.floor(window.innerHeight / 2) - 100,
  },
  hasSound: false,
};
