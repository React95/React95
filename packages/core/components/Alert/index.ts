import { Alert } from './Alert';

export default Alert;
