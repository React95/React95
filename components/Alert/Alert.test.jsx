import React from 'react';
import Alert from './Alert';
import { render, fireEvent } from 'react-testing-library';

describe('<Alert />', () => {
  describe('Snapshots', () => {
    it('should match snapshot with error type', () => {
      const { container } = render(
        <Alert title="Error" type="error" message="Error" />
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with warning type', () => {
      const { container } = render(
        <Alert title="Warning" type="warning" message="Warning" />
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with info type', () => {
      const { container } = render(
        <Alert title="Info" type="info" message="Info" />
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe('closeAlert prop', () => {
    it('should call closeAlert when modal "OK" button is pressed', () => {
      const closeAlertMock = jest.fn();
      const { getByText } = render(
        <Alert
          title="closeAlert"
          message="closeAlert"
          closeAlert={closeAlertMock}
        />
      );

      fireEvent.click(getByText('OK'));

      expect(closeAlertMock).toHaveBeenCalled();
    });
  });
});
