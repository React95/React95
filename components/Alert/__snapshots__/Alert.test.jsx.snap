// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> Snapshots should match snapshot with error type 1`] = `
.c11 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 0px #000;
}

.c11:disabled {
  color: #868a8e;
}

.c11:focus {
  outline: none;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -0.5px -0.5px 0px 1px #868a8e,1px 1px 0 1px #000;
  outline: 1px dotted #000;
  outline-offset: -5px;
}

.c11:active {
  padding: 8px 20px 5px;
  outline: 1px dotted #000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e,0 0 0 1px #000;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  width: autopx;
  height: 120px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 1px #000;
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  background-color: #00007f;
  color: white;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c3 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 0px #000;
  margin-right: 2px;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-size: 10px;
}

.c5:disabled {
  color: #868a8e;
}

.c5:focus {
  outline: none;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -0.5px -0.5px 0px 1px #868a8e,1px 1px 0 1px #000;
  outline: 1px dotted #000;
  outline-offset: -5px;
}

.c5:active {
  padding: 8px 20px 5px;
  outline: 1px dotted #000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e,0 0 0 1px #000;
}

.c5:last-child {
  margin-right: 0;
}

.c5:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c5:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -1px -1px 0px 1px #868a8e;
}

.c6 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 6px 6px 6px;
}

.c10 .c4 {
  margin-right: 6px;
  min-width: 70px;
}

.c10 .c4:last-child {
  margin-right: 0;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c8 {
  background-image: url(test-file-stub);
  background-repeat: no-repeat;
  background-size: 70%;
  height: 45px;
  width: 45px;
  background-position: center;
  margin-right: 10px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
  <div
    class="react-draggable c0"
    height="120"
    style="transform: translate(362px,304px);"
    width="auto"
  >
    <div
      class="draggable c1"
    >
      
      <div
        class="c2"
      >
        Error
      </div>
      <ul
        class="c3"
      >
        <button
          class="c4 c5"
        >
          ?
        </button>
        <button
          class="c4 c5"
        >
          x
        </button>
      </ul>
    </div>
    <div
      class="c6"
    >
      <div
        class="c7"
      >
        <div
          class="c8"
          type="error"
        />
        <div
          class="c9"
        >
          Error
        </div>
      </div>
    </div>
    <div
      class="c10"
    >
      <button
        class="c4 c11"
        value="OK"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;

exports[`<Alert /> Snapshots should match snapshot with info type 1`] = `
.c11 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 0px #000;
}

.c11:disabled {
  color: #868a8e;
}

.c11:focus {
  outline: none;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -0.5px -0.5px 0px 1px #868a8e,1px 1px 0 1px #000;
  outline: 1px dotted #000;
  outline-offset: -5px;
}

.c11:active {
  padding: 8px 20px 5px;
  outline: 1px dotted #000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e,0 0 0 1px #000;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  width: autopx;
  height: 120px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 1px #000;
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  background-color: #00007f;
  color: white;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c3 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 0px #000;
  margin-right: 2px;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-size: 10px;
}

.c5:disabled {
  color: #868a8e;
}

.c5:focus {
  outline: none;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -0.5px -0.5px 0px 1px #868a8e,1px 1px 0 1px #000;
  outline: 1px dotted #000;
  outline-offset: -5px;
}

.c5:active {
  padding: 8px 20px 5px;
  outline: 1px dotted #000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e,0 0 0 1px #000;
}

.c5:last-child {
  margin-right: 0;
}

.c5:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c5:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -1px -1px 0px 1px #868a8e;
}

.c6 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 6px 6px 6px;
}

.c10 .c4 {
  margin-right: 6px;
  min-width: 70px;
}

.c10 .c4:last-child {
  margin-right: 0;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c8 {
  background-image: url(test-file-stub);
  background-repeat: no-repeat;
  background-size: 70%;
  height: 45px;
  width: 45px;
  background-position: center;
  margin-right: 10px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
  <div
    class="react-draggable c0"
    height="120"
    style="transform: translate(362px,304px);"
    width="auto"
  >
    <div
      class="draggable c1"
    >
      
      <div
        class="c2"
      >
        Info
      </div>
      <ul
        class="c3"
      >
        <button
          class="c4 c5"
        >
          ?
        </button>
        <button
          class="c4 c5"
        >
          x
        </button>
      </ul>
    </div>
    <div
      class="c6"
    >
      <div
        class="c7"
      >
        <div
          class="c8"
          type="info"
        />
        <div
          class="c9"
        >
          Info
        </div>
      </div>
    </div>
    <div
      class="c10"
    >
      <button
        class="c4 c11"
        value="OK"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;

exports[`<Alert /> Snapshots should match snapshot with warning type 1`] = `
.c11 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 0px #000;
}

.c11:disabled {
  color: #868a8e;
}

.c11:focus {
  outline: none;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -0.5px -0.5px 0px 1px #868a8e,1px 1px 0 1px #000;
  outline: 1px dotted #000;
  outline-offset: -5px;
}

.c11:active {
  padding: 8px 20px 5px;
  outline: 1px dotted #000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e,0 0 0 1px #000;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  width: autopx;
  height: 120px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 1px #000;
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  background-color: #00007f;
  color: white;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c3 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 0px #000;
  margin-right: 2px;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-size: 10px;
}

.c5:disabled {
  color: #868a8e;
}

.c5:focus {
  outline: none;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -0.5px -0.5px 0px 1px #868a8e,1px 1px 0 1px #000;
  outline: 1px dotted #000;
  outline-offset: -5px;
}

.c5:active {
  padding: 8px 20px 5px;
  outline: 1px dotted #000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e,0 0 0 1px #000;
}

.c5:last-child {
  margin-right: 0;
}

.c5:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c5:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -1px -1px 0px 1px #868a8e;
}

.c6 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 6px 6px 6px;
}

.c10 .c4 {
  margin-right: 6px;
  min-width: 70px;
}

.c10 .c4:last-child {
  margin-right: 0;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c8 {
  background-image: url(test-file-stub);
  background-repeat: no-repeat;
  background-size: 70%;
  height: 45px;
  width: 45px;
  background-position: center;
  margin-right: 10px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
  <div
    class="react-draggable c0"
    height="120"
    style="transform: translate(362px,304px);"
    width="auto"
  >
    <div
      class="draggable c1"
    >
      
      <div
        class="c2"
      >
        Warning
      </div>
      <ul
        class="c3"
      >
        <button
          class="c4 c5"
        >
          ?
        </button>
        <button
          class="c4 c5"
        >
          x
        </button>
      </ul>
    </div>
    <div
      class="c6"
    >
      <div
        class="c7"
      >
        <div
          class="c8"
          type="warning"
        />
        <div
          class="c9"
        >
          Warning
        </div>
      </div>
    </div>
    <div
      class="c10"
    >
      <button
        class="c4 c11"
        value="OK"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;
